### DISCUSSION TITLE
posts/2024-04-25-grammar-as-a-biometric-for-authorship-verification/

### DISCUSSION THREAD

**Discussion by giscus** [vgherard/vgherard.github.io#7] [Comments]
# posts/2024-04-25-grammar-as-a-biometric-for-authorship-verification/

Notes on preprint 2403.08462 by A. Nini, O. Halvani, L. Graner, S. Ishihara 
and myself.

https://vgherard.github.io/posts/2024-04-25-grammar-as-a-biometric-for-authorship-verification/

**Comment by under-score** [comment]
Could you please add a link to extract_grammar() ?

**Reply by vgherard** [reply]
Hi, the implementation of that function is not totally trivial, requiring in particular a good part-of-speech tagging algorithm... which is why I'm not giving the details here. That said:

* With the published version of this article, we're planning to attach the actual code we used to test the model. This will also include a (standalone) implementation of the content masking algorithm.
* Meanwhile, you could have a look at my collaborator's work-in-progress R package [idiolect](https://github.com/andreanini/idiolect). The function `idiolect::contentmask()` is our `extract_grammar()`. You can have a look at this function's code to get some grasp on what the grammar extraction consists of.

Hope this helps :-)

**Comment by under-score** [comment]
million thanks - agree immediately that this is far from being trivial.
My use case is the identification of thesis writing offices and paper mills ;-)

**Reply by vgherard** [reply]
You are welcome :-) 

Interesting application! Andrea, the author of [idiolect](https://github.com/andreanini/idiolect), is a forensic scientist, so you may find some useful stuff in his library.
